src/intrinsics/prepack/global.js

```diff
@@ -118,6 +119,38 @@ export default function(realm: Realm): void {
     configurable: true,
   });
 
+  global.$DefineOwnProperty("__additionalFunctions", {
+    value: new ObjectValue(realm, realm.intrinsics.ObjectPrototype, "__additionalFunctions", true),
+    writable: true,
+    enumerable: false,
+    configurable: true,
+  });
+
+  global.$DefineOwnProperty("__registerAdditionalFunction", {
+    value: new NativeFunctionValue(
+      realm,
+      "global.__registerAdditionalFunction",
+      "__registerAdditionalFunction",
+      0,
+      (context, [functionValue]) => {
+        invariant(functionValue instanceof ECMAScriptSourceFunctionValue);
```

Comment:
`__registerAdditionalFunction` may be called from a partially-evaluate context whose effects are later abandoned. And yet, `__registerAdditionalFunction` stores the given function value in a global map `recordedAdditionalFunctions` that doesn't participate in effect tracking / undoing.

What are the implications of this? A comment and/or test case would be appropriate. Reading all the reason, I suspect that the result is just a really bad error message when the indexed global variable does not hold a plain function value in the end, but no horrible crash. In that case, a little bit of work should be done to make the error message / diagnosis tolerable.